### PR TITLE
chore: Dependency Updates

### PR DESCRIPTION
**Pull request by refcell**
### Description

Updates dependencies, removing the crates.io patch.

Re-organizes the workspace toml, making dependencies more easily accessible.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,10 +23,109 @@ default-members = ["bin/hera"]
...
+reth-execution-types = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-exex = { git = "https://github.com/paradigmxyz/reth", features = ["serde"], rev = "000b412" }
+reth-network-peers = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-node-api = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-node-ethereum = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-primitives = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-provider = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-revm = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-evm = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
+reth-tracing = { git = "https://github.com/paradigmxyz/reth", rev = "000b412" }
```
Comment (refcell) [inline Cargo.toml:87]: Not a huge fan of using revisions here, but there isn't a tagged version with the most revent alloy versions we need.

File: Cargo.toml [hunk 2]
```diff
@@ -23,10 +23,109 @@ default-members = ["bin/hera"]
...
+# Workspace
+op-net = { path = "crates/net" }
+kona-providers = { path = "crates/kona-providers" }
+rollup = { path = "crates/rollup" }
+ser = { path = "crates/ser" }
+
+# Optimism
+superchain = { version = "0.5", default-features = false }
+kona-primitives = { git = "https://github.com/anton-rs/kona", branch = "main", default-features = true }
+kona-derive = { git = "https://github.com/anton-rs/kona", branch = "main", default-features = true }
```
Comment (merklefruit) [inline Cargo.toml:43]: would love to pin these when there is a new release also!

**Review by merklefruit — APPROVED** [review]
Lovely, much needed cleanup!

File: Cargo.toml (hunk 1 above)
Comment (merklefruit) [reply Cargo.toml:87]: I'm sure they'll make a new one soon, it's been almost a month!

**Pull request merged**